trace/http_test.go

```diff
@@ -2,73 +2,90 @@ package trace
 import (
 	"context"
+	"fmt"
 	"net/http"
 	"net/http/httptest"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 )
 
-func TestGetTraceFromHTTRequest(t *testing.T) {
+func TestGetTraceFromHTTPRequest(t *testing.T) {
 	r, err := http.NewRequestWithContext(context.Background(), "POST", "URL", nil)
 	assert.NoError(t, err)
 
 	r.Header.Add(headerTraceID, "00000000000000000000000000000019")
 	r.Header.Add(headerProbabilitySample, "0.5")
 
-	trace := GetTraceFromHTTRequest(r)
+	trace := GetTraceFromHTTPRequest(r)
 
 	assert.Equal(t, "00000000000000000000000000000019", trace.ID.String())
 	assert.Equal(t, 0.5, *trace.ProbabilitySample)
 }
 
-func TestGetTraceFromHTTRequest_ErrorParseProbabilitySample(t *testing.T) {
+func TestGetTraceFromHTTPRequest_ErrorParseProbabilitySample(t *testing.T) {
 	r, err := http.NewRequestWithContext(context.Background(), "POST", "URL", nil)
 	assert.NoError(t, err)
 
 	r.Header.Add(headerTraceID, "00000000000000000000000000000019")
 	r.Header.Add(headerProbabilitySample, "0.5a")
 
-	trace := GetTraceFromHTTRequest(r)
+	trace := GetTraceFromHTTPRequest(r)
 
 	assert.Equal(t, "00000000000000000000000000000019", trace.ID.String())
 	assert.Nil(t, trace.ProbabilitySample)
 }
 
-func TestGetTraceFromHTTRequest_WithoutHeader(t *testing.T) {
+func TestGetTraceFromHTTPRequest_WithoutHeader(t *testing.T) {
 	r, err := http.NewRequestWithContext(context.Background(), "POST", "URL", nil)
 	assert.NoError(t, err)
 
-	trace := GetTraceFromHTTRequest(r)
+	trace := GetTraceFromHTTPRequest(r)
 
 	assert.True(t, IDIsEmpty(trace.ID))
 	assert.Nil(t, trace.ProbabilitySample)
 }
 
-func TestGetTraceFromHTTRequest_WithoutProbabilitySample(t *testing.T) {
+func TestGetTraceFromHTTPRequest_WithoutProbabilitySample(t *testing.T) {
 	r, err := http.NewRequestWithContext(context.Background(), "POST", "URL", nil)
 	assert.NoError(t, err)
 
 	r.Header.Add(headerTraceID, "00000000000000000000000000000019")
 
-	trace := GetTraceFromHTTRequest(r)
+	trace := GetTraceFromHTTPRequest(r)
 
 	assert.Equal(t, "00000000000000000000000000000019", trace.ID.String())
 	assert.Nil(t, trace.ProbabilitySample)
 }
 
-func TestGetTraceFromHTTRequest_WithoutTraceID(t *testing.T) {
+func TestGetTraceFromHTTPRequest_WithoutTraceID(t *testing.T) {
 	r, err := http.NewRequestWithContext(context.Background(), "POST", "URL", nil)
 	assert.NoError(t, err)
 
 	r.Header.Add(headerProbabilitySample, "0.5")
 
-	trace := GetTraceFromHTTRequest(r)
+	trace := GetTraceFromHTTPRequest(r)
 
 	assert.True(t, IDIsEmpty(trace.ID))
 	assert.Equal(t, 0.5, *trace.ProbabilitySample)
 }
 
+func TestGetTraceFromHTTPResponse(t *testing.T) {
+	ps := 0.75
+	trace := Trace{
+		ID:                decode([]byte("000000000000000000000000bebacafe")),
+		ProbabilitySample: &ps,
+	}
+
+	response := http.Response{}
+	response.Header = make(http.Header)
+	response.Header.Set(headerTraceID, trace.ID.String())
+	response.Header.Set(headerProbabilitySample, fmt.Sprintf("%f", *trace.ProbabilitySample))
```

Comment:
Seeing this makes me think if we should add some function to trace package to get string forms of id and probabilitysample. And also a way to parse it.

Reply:
These methods already exists for `ID` (decode and string). For probability sample it would be nice, since its cumbersome to handle float pointers.